alacritty_terminal/src/ansi.rs

```diff
@@ -974,13 +974,21 @@ where
             b"4" => {
                 if params.len() > 1 && params.len() % 2 != 0 {
                     for chunk in params[1..].chunks(2) {
-                        let index = parse_number(chunk[0]);
-                        let color = xparse_color(chunk[1]);
-                        if let (Some(i), Some(c)) = (index, color) {
-                            self.handler.set_color(i as usize, c);
-                            return;
+                        if let Some(index) = parse_number(chunk[0]) {
```

Comment:
Really not a fan of the additional nesting here, especially because it already was pretty bad.